source/gems/dry-core/index.html.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Introduction &amp; Usage
+description: A toolset of small support modules used throughout the @dry-rb & @rom-rb ecosystems
+layout: gem-single
+order: 5
+type: gem
+name: dry-core
+---
+
+`dry-core` is a simple toolset that can be used in many places.
+
+## Usage
+
+### Cache
+Allows you to cache call results that are solely determined by arguments.
+
+```ruby
+require 'dry/core/cache'
+
+class Foo
+  extend Dry::Core::Cache
+
+  def heavy_computation(arg1, arg2)
+    fetch_or_store(arg1, arg2) { arg1 ^ arg2 }
+  end
+end
+```
+
+### Class Attributes
+
+```ruby
+require 'dry/core/class_attributes'
+
+class MyClass
+  extend Dry::Core::ClassAttributes
+
+  defines :one, :two
+
+  one 1
+  two 2
+end
+
+class OtherClass < MyClass
+  two 'two'
+end
+
+MyClass.one # => 1
+MyClass.two # => 2
+
+OtherClass.one # => 1
+OtherClass.two # => 'two'
+```
+
+### Class Builder
+
+```ruby
+require 'dry/core/class_builder'
+
+builder = Dry::Core::ClassBuilder.new(name: 'MyClass')
+
+klass = builder.call
+klass.name # => "MyClass"
+```
+
+### Constants
+A list of constants you can use to avoid memory allocations or identity checks.
+
+* `EMPTY_ARRAY`
+* `EMPTY_HASH`
+* `EMPTY_OPTS`
+* `EMPTY_SET`
+* `EMPTY_STRING`
+* `Undefined` - A special value you can use as a default to know if no arguments were passed to you method
+
+### Deprecations
+
+Fox deprecate ruby methods you need to extend `Dry::Core::Deprecations` module
```

Comment:
Not sure what `Fox deprecate ruby methods` means here? Should it be `For deprecating ruby methods, ...`?